### PR TITLE
[#IOAPE-434] Added FeatureFlag for cms to legacy sync test

### PR DESCRIPTION
**Pull request by giamma1295**
Added a Feature Flag in order to selectively allow user to Sync Services from CMS to Legacy.
#### List of Changes
- Add `USERID_CMS_TO_LEGACY_SYNC_INCLUSION_LIST`
- Add `feature-flag-handler.ts` to manage feature flags
- Update Service History Watcher to use the new feature flag

#### How Has This Been Tested?
Unit Tests

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  getSubscription,
+  parseOwnerIdFullPath,
+} from "../lib/clients/apim-client";
+
+const wildcard = "*" as NonEmptyString;
+
+// Return True if the element is in the list or if the list contains the wildcard
+const isElementAllowedOnList =
+  (list: ReadonlyArray<NonEmptyString>) => (element: NonEmptyString) =>
+    list.indexOf(wildcard) >= 0 || list.indexOf(element) >= 0;
```
Comment (gunzip) [inline apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts:17]:
```suggestion
    list.include(swildcard) || list.includes(element);
```

File: apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    ),
+    TE.mapLeft(
+      (_) =>
+        new Error(
+          `An error has occurred while retrieving service '${serviceId}'`
+        )
+    ),
+    TE.map((serviceSubscription) =>
+      pipe(
+        serviceSubscription.ownerId as NonEmptyString,
```
Comment (gunzip) [inline apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts:46]:
```suggestion
    TE.map(({ ownerId }) =>
      pipe(
        ownerId as NonEmptyString,
```

File: apps/io-services-cms-webapp/src/watchers/on-service-history-change.ts [hunk 3]
```diff
@@ -101,15 +104,28 @@ const toRequestSyncLegacyAction = (
...
-  NoAction | RequestSyncLegacyAction
-> = ({ item }) =>
-  pipe(
-    item,
-    O.fromPredicate((itm) => itm.fsm.lastTransition !== SYNC_FROM_LEGACY),
-    O.map(toRequestSyncLegacyAction),
-    O.getOrElse(() => noAction),
-    E.right
-  );
+export const handler =
```
Comment (gunzip) [inline apps/io-services-cms-webapp/src/watchers/on-service-history-change.ts:107]: why not just:

`ReaderEither<{ item: ServiceHistory, config: IConfig, apimClient: ApiManagementClient}, ...>`

in this way you don't have to pass around the apimClient.

Comment (giamma1295) [reply apps/io-services-cms-webapp/src/watchers/on-service-history-change.ts:107]: Cannot be dono atm, cause we need to edit processBatchOf which is used for all Handler CosmosDbTrigger

File: apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts (hunk 2 above)
Comment (giamma1295) [reply apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts]: 089d886

File: infra/src/key_vault.tf [hunk 4]
```diff
@@ -33,6 +33,11 @@ data "azurerm_key_vault_secret" "serviceid_quality_check_exclusion_list" {
   key_vault_id = module.key_vault_domain.id
 }
 
+data "azurerm_key_vault_secret" "userid_cms_to_legacy_sync_inclusion_list" {
+  name         = "USERID-CMS-TO-LEGACY-SYNC-INCLUSION-LIST"
+  key_vault_id = module.key_vault_domain.id
+}
+
```
Comment (rodoviti) [inline infra/src/key_vault.tf:40]: why a list of user id is a secret? I mean, I'm pretty sure they are not sensitive datas, so there is no need to put them in the key vault

Comment (giamma1295) [reply infra/src/key_vault.tf]: bc9c699

File: apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * @param config
+ * @param apimClient
+ * @param serviceId
+ * @returns
+ */
+export const isUserEnabledForCmsToLegacySync = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  serviceId: NonEmptyString
+) =>
```
Comment (rodoviti) [inline apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts:30]: add return type to "enforce" type checking

Comment (giamma1295) [reply apps/io-services-cms-webapp/src/utils/feature-flag-handler.ts]: f5fb2fb

**Review by rodoviti — APPROVED** [review]

**Pull request merged**